packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -51,21 +65,59 @@ export class SharedTreeCore<TChangeRebaser extends ChangeRebaser<any, any, any>>
         this.rebaser = new Rebaser(changeRebaser);
         this.localState = this.rebaser.empty;
         this.sequencedRevision = this.rebaser.empty;
+
+        this.summaryElements = indexes.map((i) => i.summaryElement).filter((e): e is SummaryElement => e !== undefined);
+        assert(
+            new Set(this.summaryElements.map((e) => e.key)).size === this.summaryElements.length,
+            "Index summary element keys are not unique",
```

Comment:
Sometimes its unclear in an assert message if the message describes what must be true, or is a description of what went wrong.

We can make this one unambiguous:
```suggestion
            "Index summary element keys must be unique",
```

Reply:
Sure thing, good idea. Ideally we can be consistent with our error messages. I'll start preferring to write the invariant that is being broken rather than stating that it is broken, as you are suggesting here.